sound/soc/sof/topology.c

```diff
@@ -1065,6 +1065,46 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 	return 0;
 }
 
+static void sof_disconnect_dai_widget(struct snd_soc_component *scomp,
+				      struct snd_soc_dapm_widget *w)
+{
+	struct snd_soc_card *card = scomp->card;
+	struct snd_soc_pcm_runtime *rtd;
+	struct snd_soc_dai *cpu_dai;
+	int i;
+
+	list_for_each_entry(rtd, &card->rtd_list, list) {
+
+		if (!w->sname || !rtd->dai_link->stream_name)
```

Comment:
the `if (!w->sname)` can be moved before the big loop, it is not going to magically become non NULL.

Reply:
it's the same loop as in the 'register' case, so I just copy-pasted. we can change the loop but we have to do it consistently in the register and unregister cases